library/src/main/java/com/datatorrent/lib/io/fs/AbstractSingleFileOutputOperator.java

```diff
@@ -52,6 +81,7 @@ protected String getFileName(INPUT tuple)
   public void setOutputFileName(String outputFileName)
   {
     this.outputFileName = outputFileName;
+    partitionedFileName = String.format(partitionedFileNameformat, outputFileName, physicalPartitionId);
```

Comment:
This assumption is correct, my suggestion then is: Has this capability been tested with an actual application on a cluster (not just local mode) where the file name is changed mid-stream ? If so, suggest we document that this feature is supported and add a unit test that tests it.